tools/runners/run-unittests.py

```diff
@@ -0,0 +1,98 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+import argparse
+import glob
+import os
+import subprocess
+import sys
+
+TERM_NORMAL = '\033[0m'
+TERM_RED = '\033[1;31m'
+TERM_GREEN = '\033[1;32m'
+
+def get_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-q', '--quiet', action='store_true',
+                        help='Only print out failing tests')
+    parser.add_argument('path',
+                        help='Path of test binaries')
+
+    script_args = parser.parse_args()
+    return script_args
+
+
+def get_unittests(path):
+    unittests = []
+    files = glob.glob(os.path.join(path, 'unit-*'))
+    for testfile in files:
+        if os.path.isfile(testfile) and os.access(testfile, os.X_OK):
+            if sys.platform != 'win32' or testfile.endswith(".exe"):
+                unittests.append(testfile)
+    unittests.sort()
+    return unittests
+
+
+def main(args):
+    unittests = get_unittests(args.path)
+    total = len(unittests)
+    if total == 0:
+        print("%s: no unit-* test to execute", args.path)
+        return 1
+
+    tested = 0
+    passed = 0
+    failed = 0
+
+    runtime = os.environ.get('RUNTIME')
+    test_cmd = [runtime] if runtime else []
+
+    for test in unittests:
+        tested += 1
+        testpath = os.path.relpath(test)
+        proc = subprocess.Popen(test_cmd + [test],
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.STDOUT,
+                                universal_newlines=True)
+        output = proc.communicate()[0]
+        if proc.returncode == 0:
+            passed += 1
+            if not args.quiet:
+                print("[%4d/%4d] %sPASS: %s%s" % (tested, total, TERM_GREEN, testpath, TERM_NORMAL))
+        else:
+            failed += 1
+            print("[%4d/%4d] %sFAIL (%d): %s%s" % (tested, total, TERM_RED, proc.returncode, testpath, TERM_NORMAL))
+            print("================================================")
+            print(output)
+            print("================================================")
+
+    print("\n[summary] %s\n" % os.path.join(os.path.relpath(args.path), "unit-*"))
+    print("TOTAL: %d" % total)
+    print("%sPASS: %d%s" % (TERM_GREEN, passed, TERM_NORMAL))
+    print("%sFAIL: %d%s\n" % (TERM_RED, failed, TERM_NORMAL))
+
+    success_color = TERM_GREEN if passed == total else TERM_RED
+    print("%sSuccess: %d%%%s" % (success_color, passed*100/total, TERM_NORMAL))
+
+    if failed > 0:
+        return 1
+
+    return 0
```

Comment:
Do we need the return values at all? In this code the return value is not used at all.

Reply:
Good catch, we need it, but I missed to return this rc later. Fixed.